### ISSUE TITLE
C# client throws StatusCode=Unknown Detail="Stream removed"

### ISSUE THREAD

**Issue by chinadragon0515** [grpc/grpc#22462]
I created a simple grpc service with guide from https://docs.microsoft.com/en-us/aspnet/core/grpc/aspnetcore?view=aspnetcore-3.1&tabs=visual-studio

Then I write a client with .NET core to call the service, it works well.
The code looks like this, 

            var channel = GrpcChannel.ForAddress("https://localhost:5001");
            var client = new Greeter.GreeterClient(channel);
            var reply = await client.SayHelloAsync(new HelloRequest { Name = "GreeterClient" });
And reference packages are 
Grpc.Net.Client" Version="2.27.0" 
Google.Protobuf" Version="3.11.4" 
Grpc" Version="2.27.0" 
Grpc.Tools" Version="2.27.0" 

Then I write a client with .NET framework, the code looks like below,

            Channel channel = new Channel("localhost:5001", ChannelCredentials.Insecure);
            var client = new Greeter.GreeterClient(channel);
            var reply = client.SayHello(new HelloRequest { Name = "GreeterClient" });

And package references are as below,
Grpc" Version="2.27.0" 
Google.Protobuf" Version="3.11.4" 
Grpc" Version="2.27.0" 
Grpc.Tools" Version="2.27.0" 

I always got exception when use .NET framework client and exception is as below, 
Grpc.Core.RpcException
  HResult=0x80131500
  Message=Status(StatusCode=Unknown, Detail="Stream removed")
  Source=mscorlib
  StackTrace:
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Grpc.Core.Internal.AsyncCall`2.UnaryCall(TRequest msg)
   at Grpc.Core.DefaultCallInvoker.BlockingUnaryCall[TRequest,TResponse](Method`2 method, String host, CallOptions options, TRequest request)
   at Grpc.Core.Interceptors.InterceptingCallInvoker.<BlockingUnaryCall>b__3_0[TRequest,TResponse](TRequest req, ClientInterceptorContext`2 ctx)
   at Grpc.Core.ClientBase.ClientBaseConfiguration.ClientBaseConfigurationInterceptor.BlockingUnaryCall[TRequest,TResponse](TRequest request, ClientInterceptorContext`2 context, BlockingUnaryCallContinuation`2 continuation)
   at Grpc.Core.Interceptors.InterceptingCallInvoker.BlockingUnaryCall[TRequest,TResponse](Method`2 method, String host, CallOptions options, TRequest request)
   at Helloworld.Greeter.GreeterClient.SayHello(HelloRequest request, CallOptions options) in E:\dev\vs\SHIRGrpc\ShirGrpcNetFrameworkContract\obj\Debug\netstandard2.0\HelloworldGrpc.cs:line 109
   at Helloworld.Greeter.GreeterClient.SayHello(HelloRequest request, Metadata headers, Nullable`1 deadline, CancellationToken cancellationToken) in E:\dev\vs\SHIRGrpc\ShirGrpcNetFrameworkContract\obj\Debug\netstandard2.0\HelloworldGrpc.cs:line 99
   at ShirGrpcNetFrameworkClient.Program.Main(String[] args) in E:\dev\vs\SHIRGrpc\ShirGrpcNetFrameworkClient\Program.cs:line 21

**Comment by ctclements** [comment]
Wow, I am really glad you posted this.  I am seeing this exact same issue, same packages and versions as you.  I spent all day yesterday trying to figure out what was going on without any luck.

**Comment by tchrikch** [comment]
same issue here - I can reproduce it going back to 2.23 on Grpc packages in .Net Framework.

**Comment by chinadragon0515** [comment]
I found the issue is my GRPC server is enabled with default development certificate, and it requires SSL.
I write code like below, it resolves the issue, but I met a new issue that server certificate is not trusted even I have added server certificate into windows trust store. I opened a new issue for this. 
var secureChanel = new SslCredentials();
Channel channel = new Channel("host:5001", secureChanel);

As this is not code bug, but configuration issue of client code, I close this issue.
